fix(panel): update edit menu on focus

pinussilvestrus opened the pull request:
Closes #2716

------

Artifacts to try it out

- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/2716-copy-paste-linting/camunda-modeler-2716-copy-paste-linting-linux-x64.tar.gz
- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/2716-copy-paste-linting/camunda-modeler-2716-copy-paste-linting-mac.dmg
- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/2716-copy-paste-linting/camunda-modeler-2716-copy-paste-linting-mac.zip
- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/2716-copy-paste-linting/camunda-modeler-2716-copy-paste-linting-win-ia32.zip
- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/2716-copy-paste-linting/camunda-modeler-2716-copy-paste-linting-win-x64.zip

pinussilvestrus commented:
I'd need some volunteers to double-check copy & paste still works on Windows & Linux (via [using the artifacts](https://github.com/camunda/camunda-modeler/pull/2809#issue-1168079492))

* Windows @Skaiir @philippfromme?
* Linux @MaxTru @marstamm?

marstamm commented:
I'll take care of linux

marstamm commented:
I can copy the message using ctrl+c/ctrl+v. The activity ID is not copied. Is that expected?

(`Element of type <bpmn:Task> not supported by Camunda Cloud 1.3`)

pinussilvestrus commented:
I can confirm that on Mac. But seems like this is on purpose (?) since only the message is selectable, not the full linting issue (cf. https://github.com/camunda/camunda-modeler/pull/2696/files). 

![image](https://user-images.githubusercontent.com/9433996/158143364-11e6ce4b-c9d5-4ddb-8fbb-3dcf1d692b6c.png)

Skaiir commented:
Having a look

Skaiir commented:
Working 👍

andreasgeier commented:
> The activity ID is not copied. Is that expected?
>> But seems like this is on purpose (?) since only the message is selectable, not the full linting issue

I'd consider that as part of the bug. How would a copied list like 

```
Element of type <bpmn:Task> not supported by Camunda Cloud 1.3
Element of type <bpmn:Task> not supported by Camunda Cloud 1.3
Element of type <bpmn:Task> not supported by Camunda Cloud 1.3
```
with no reference for the multiple messages be useful?

pinussilvestrus commented:
Thanks for clarifying, I was not aware of that (since it was even implemented this way). I will fix that.

andreasgeier commented:
We should copy the message exactly as they are shown in the panel:

`Error : Activity_123456 - Element of type <bpmn:Task> not supported by Camunda Cloud 1.3`

pinussilvestrus commented:
I adjusted it accordingly: https://github.com/camunda/camunda-modeler/pull/2809/commits/8609fe9b196aba95754374eb8479e2649fc6912d. The Artifacts are also updated. 

Thanks for mentioning @andreasgeier!

andreasgeier reviewed (approved):
✅  LGTM

pinussilvestrus commented:
@smbea @Skaiir any chance to get a review here, soon? 🙂  If not I could assign someone other

Skaiir commented:
@pinussilvestrus A bit swamped at the moment so I would defer to Bea on this one.

smbea commented:
I will look into this today

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/camunda/camunda-modeler?pullRequest=2809) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/camunda/camunda-modeler?pullRequest=2809) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/camunda/camunda-modeler?pullRequest=2809) it.</sub>

smbea commented:
If I copy the whole message the ID is copied, but if I try to copy only the ID it doesn't work

pinussilvestrus commented:
Indeed, simply copy & paste the activity id is still broken.

![Kapture 2022-03-22 at 13 01 52](https://user-images.githubusercontent.com/9433996/159477652-843c5d50-8125-4684-b80c-f81e98a3ed2c.gif)

I will investigate this, thanks!

pinussilvestrus commented:
Root cause: when clicking + selecting the element id, the click event fires the action to select the element on the canvas, which moves the focus from the selected id to the element. Therefore, it will try to copy the element, not the selected text.

![Kapture 2022-03-22 at 13 15 13](https://user-images.githubusercontent.com/9433996/159480141-f32b1eb7-5451-4e49-ac9c-55f3b81b69c8.gif)

Since this it's the expected behavior to select the element, I don't know whether we should do something about it. There might be hacky workarounds, e.g. [to prevent the onClick on selecting the text](https://stackoverflow.com/a/31982530), though. Wdydt @andreasgeier @smbea?

andreasgeier commented:
I have no insights whether users want to copy just a single activity ID. So, I'd say if it is a small effort to enable it, let's do it. Otherwise, let's wait for complaints.

smbea commented:
Then, all good to go on my end
